notebooks/02-ingestion/04-table-datatypes.ipynb

```diff
@@ -0,0 +1,851 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ea07299d-27e2-407d-8df7-2bee1127d444",
+   "metadata": {},
+   "source": [
+    "In the table above, the `DayofMonth_Filter` column is recognized as a VARCHAR because of the CAST function.\n",
+    "\n",
+    "Since Druid automatically creates [bitmap indexes](https://druid.apache.org/docs/latest/ingestion/ingestion-spec#dimension-objects) for STRING-type columns at ingestion time, it can be useful to create VARCHAR versions of columns that will be used for filtering in addition to any CLUSTERED BY column, especially when that column will never be used for any sort of mathematical calculation.\n",
+    "\n",
+    "Run the following cell to see a simple query that makes use of this filtering column."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6a116797-2dfb-4273-a50e-fda337bb3f7e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "    COUNT(*) FILTER (WHERE \"DayofMonth_Filter\" = '10') AS \"10thDayCount\",\n",
+    "    COUNT(*) FILTER (WHERE \"DayofMonth_Filter\" = '9') AS \"9thDayCount\"\n",
+    "FROM \"example-flights-types-4\"\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "143b76e0-7c6a-4766-927c-4fb6ec82656f",
+   "metadata": {},
+   "source": [
+    "Building on the previous example, run the following cell to see a CAST using an internally stored string to compare with a numeric value:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "615fe582-ca47-4602-a1a6-383a0c4ca2bb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "    COUNT(*) FILTER (WHERE \"DayofMonth_Filter\" = '10') AS \"10thDayCount\",\n",
+    "    COUNT(*) FILTER (WHERE \"DayofMonth_Filter\" = '9') AS \"9thDayCount-1\",\n",
+    "    COUNT(*) FILTER (WHERE CAST(\"DayofMonth_Filter\" AS BIGINT) = 9) AS \"9thDayCount-2\",\n",
+    "    COUNT(*) FILTER (WHERE CAST(\"DayofMonth_Filter\" AS BIGINT) = 8) AS \"8thDayCount\"\n",
+    "FROM \"example-flights-types-4\"\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4f42e716-592a-42aa-bfa2-d743f963983f",
+   "metadata": {},
+   "source": [
+    "You can also convert from a TIMESTAMP to a string.\n",
+    "\n",
+    "Run the following cell to see a CAST from TIMESTAMP to a string, as well as an example of the TIME_FORMAT function to create the same output.\n",
+    "\n",
+    "> For more examples of TIME_FORMAT check out the dedicated notebook on [date and time functions](../03-query/07-functions-datetime.ipynb)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f2cbad55-b547-42e1-8713-6e1f6144c1b2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
```

Comment:
Add 'a' to the time format to make the results less confusing.... it seems like it is shifting time by 12 hours but it is really just missing the AM, PM... like this: 
```
SELECT
    CAST("__time" AS VARCHAR) AS "__time_VARCHAR",
    TIME_FORMAT(__time, 'YYYY-MM-dd hh:mm:ss a') AS "__time_FORMAT"
FROM "example-flights-types-4"
WHERE TIME_IN_INTERVAL("__time",'2005-11-08/PT10M')
```

Reply:
Ah yes lovely!